### PR TITLE
Added features and fixes to screencast mode

### PR DESCRIPTION
**Pull request by webarthur**
* Fix annoying arrow key repetition in screencast mode
* Fix label of pressed arrow keys in screencast mode
* Add option to hide single arrow keys pressed in screencast mode

### CODE REVIEW

**Comment by webarthur** [comment]
@microsoft-github-policy-service agree [company="{your company}"]

**Comment by webarthur** [comment]
@microsoft-github-policy-service agree

File: src/vs/workbench/browser/actions/developerActions.ts [hunk 1]
```diff
@@ -276,6 +285,21 @@ class ToggleScreencastModeAction extends Action2 {
 				}
 
 				if (!configurationService.getValue('screencastMode.onlyKeyboardShortcuts') || !titleLabel || shortcut?.commandId && (format === 'keys' || format === 'commandAndKeys' || format === 'commandWithGroupAndKeys')) {
+					// Fix label for arrow keys
+					if (keyLabel) {
+						if (keyLabel.includes('UpArrow')) {
```
Comment (joaomoreno) [inline src/vs/workbench/browser/actions/developerActions.ts:290]: Are these `if` statements really needed?

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/actions/developerActions.ts (hunk 1 above)
Comment (webarthur) [reply src/vs/workbench/browser/actions/developerActions.ts:290]: I think so. I don't know how showing arrow key names can be useful:



Feel free to make any changes you think are best.

File: src/vs/workbench/browser/actions/developerActions.ts [hunk 2]
```diff
@@ -276,6 +285,21 @@ class ToggleScreencastModeAction extends Action2 {
...
+						if (keyLabel.includes('DownArrow')) {
+							keyLabel = keyLabel.replace('DownArrow', '↓');
+						}
+						if (keyLabel.includes('LeftArrow')) {
+							keyLabel = keyLabel.replace('LeftArrow', '←');
+						}
+						if (keyLabel.includes('RightArrow')) {
+							keyLabel = keyLabel.replace('RightArrow', '→');
+						}
+					}
```
Comment (joaomoreno) [inline src/vs/workbench/browser/actions/developerActions.ts:302]: This is what I meant:

```suggestion
					keyLabel = keyLabel?.replace('UpArrow', '↑')
						?.replace('DownArrow', '↓')
						?.replace('LeftArrow', '←')
						?.replace('RightArrow', '→');
```

**Review by joaomoreno — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/actions/developerActions.ts (hunk 2 above)
Comment (dominik-ba) [reply src/vs/workbench/browser/actions/developerActions.ts:302]: I'd say your suggestion looks quite fine and I would love to have this feature. the repetition is quite annoying. Please merge and add it to the next release! :)

Comment (webarthur) [reply src/vs/workbench/browser/actions/developerActions.ts:302]: @joaomoreno 

Ah, ok. Your suggested change looks better written than mine. 

What is the next step? Can you make this change or do I need to do something more?

Comment (dominik-ba) [reply src/vs/workbench/browser/actions/developerActions.ts:302]: @webarthur you should see a button "commit suggestion": https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

**Comment by dominik-ba** [comment]
So will this be merged? :)

**Comment by pbarhyl** [comment]
Also waiting for that!

**Comment by joaomoreno** [comment]
I only understood now that the setting is about not showing the arrow keys when used to move the **text editor cursor**.

**Review by joaomoreno — APPROVED** [review]

**Review by aeschli — APPROVED** [review]

**Comment by dominik-ba** [comment]
Maybe its only me, but at least I now get a pop up for each key that I type although I enabled the flag to show only on keyboard shortcuts

**Pull request merged**